src/infra/metadata/sparql/sparqlGateway.ts

```diff
@@ -0,0 +1,198 @@
+      OPTIONAL { ?generalMetadataID core:hasImage ?image . }
+
+      OPTIONAL { 
+        ?generalMetadataID core:hasLicense ?licenseResource .
+        ?licenseResource skos:prefLabel ?licenseFirstLg .
+        OPTIONAL {
+          ?licenseResource skos:prefLabel ?licenseUserLg .
+          FILTER langMatches(lang(?licenseUserLg), '${language}') .
+        }
+        OPTIONAL {
+          ?licenseResource skos:prefLabel ?licenseEnLg .
+          FILTER langMatches(lang(?licenseEnLg),'en') .
+        }
+      }
+
+      OPTIONAL { 
+        ?generalMetadataID core:hasSpatialCoverage ?areaResource .
+        ?areaResource skos:prefLabel ?spatialCoverageFirstLg .
+        OPTIONAL {
+          ?areaResource skos:prefLabel ?spatialCoverageUserLg .
+          FILTER langMatches(lang(?spatialCoverageUserLg),'${language}') .
+        }
+        OPTIONAL {
+          ?areaResource skos:prefLabel ?spatialCoverageEnLg .
+          FILTER langMatches(lang(?spatialCoverageEnLg),'en') .
+        }
+      }
+
+      OPTIONAL { 
+        ?generalMetadataID core:hasTopic ?topicResource .
+        ?topicResource skos:prefLabel ?topicFirstLg . 
+        OPTIONAL {
+          ?topicResource skos:prefLabel ?topicUserLg . 
+          FILTER langMatches(lang(?topicUserLg),'${language}') .
+        }
+        OPTIONAL {
+          ?topicResource skos:prefLabel ?topicEnLg . 
+          FILTER langMatches(lang(?topicEnLg),'en') .
+        }
+      }
+      
+      OPTIONAL { ?generalMetadataID core:hasCreator ?creator . }
+
+    
+      OPTIONAL { 
+        ?generalMetadataID core:hasFormat ?formatResource .
+        ?formatResource skos:prefLabel ?formatFirstLg . 
+        OPTIONAL {
+          ?formatResource skos:prefLabel ?formatUserLg . 
+          FILTER langMatches(lang(?formatUserLg),'${language}') .
+        }
+        OPTIONAL {
+          ?formatResource skos:prefLabel ?formatEnLg . 
+          FILTER langMatches(lang(?formatEnLg),'en') .
+        }
+      }
+
+      OPTIONAL { 
+        ?generalMetadataID core:hasTemporalCoverage ?temporalCoverage .
+        OPTIONAL {  ?temporalCoverage core:hasStartDate ?startDate . }
+        OPTIONAL {  ?temporalCoverage core:hasEndDate ?endDate . }
+      }
+
+      OPTIONAL { 
+        ?generalMetadataID core:hasCategory ?categoryResource . 
+        OPTIONAL {
+          ?categoryResource skos:prefLabel ?categoryFirstLg .
+        }
+        OPTIONAL {
+          ?categoryResource skos:prefLabel ?categoryUserLg .
+          FILTER langMatches(lang(?categoryUserLg),'${language}') .
+        }
+        OPTIONAL {
+          ?categoryResource skos:prefLabel ?categoryEnLg .
+          FILTER langMatches(lang(?categoryEnLg),'en') .
+        }
+      }
+    }
+
+    GROUP BY ?publisher ?createdBy ?updatedOn ?createdOn ?lastModifiedBy ?image ?creator ?startDate ?endDate ?tags
+    `
+
+    const mapDtoToEntity = (dto: SparqlResult): MetadataItem[] =>
+      Object.entries(dto.results.bindings[0]).map(([key, value]) => ({
```

Comment:
I would have mapped all the `dto.results.bindings` and not only the first one, so that you can handle an array with no need of using `Object.entries()`